src/arraymancer/nn/optimizers/optimizers.nim

```diff
@@ -159,20 +164,17 @@ func optimizerSGDMomentum*[M, T](model: M, learning_rate: T, momentum = T(0.0),
   ##
   ## Returns:
   ## - A SGD optimizer with momentum with the given parameters.
-  ##
-  ## Future TODO:
-  ##   Rename to ``optimize[M](model: M, OptimizerKind: typedesc[SGDMomentum], learning_rate: SomeFloat):``
-
-  # TODO: rename to optimize[M](model: M, OptimizerKind: typedesc[SGD], learning_rate: SomeFloat): ...
-  # Pending https://github.com/nim-lang/Nim/issues/7734 and https://github.com/nim-lang/Nim/issues/7733
 
   result.params = @[]
   result.lr = learning_rate
   result.momentum = momentum
   result.decay = decay
   result.nesterov = nesterov
 
-  func addLayer(result: var SgdMomentum[Tensor[T]], layer: auto) =
+proc optimizerSGDMomentum*[M, T](model: M, learning_rate: T, momentum = T(0.0), decay = T(0.0), nesterov = false): SGDMomentum[Tensor[T]] {.deprecated: "Use optimizer(model, SGDMomentum, learning_rate) instead."} =
```

Comment:
The addLayer part should be in the `optimizer` proc no?

Reply:
oops, fixed.